doc/source/contribution_and_api/unit_testing.rst

```diff
@@ -0,0 +1,171 @@
+.. _ref_unit_testing_contributing:
+
+Unit Testing
+============
+
+Unit tests are functions that validate the software by testing that the
+logic implemented inside a certain method, class or module is
+working as expected. They should try to be as atomic and 
+independent as possible.
+
+Unit testing is highly important. It enables to check that the
+changes on the code are consistent with other parts of the code
+and verify that the changes are the expected in the implementation.
+
+Unit tests are in the `tests <pymapdl_tests_>`_ directory inside this repository.
+All the tests that you will find in this directory are not
+unit tests. Some of them are integration tests. The difference between
+a unit test and an integration test is that the latter will
+test several units of the code and ensure that they work together.
+
+To verify that all the code is properly tested, we need to make sure that every piece
+of code is used (covered) in, at least one, unit test. For this purpose, in this
+repository `Codecov <codecov_>`_ tool is used to generate a coverage report of the
+committed code. It details how merging the Pull Request will impact the coverage. It
+is one of the checks that must be approved to merge the code.
+
+
+.. figure:: ../images/codecov_increase.png
+    :width: 400pt
+
+
+Coverage example
+----------------
+
+Let's show with an example how the coverage works.
+Let's suppose that we have the following library:
+
+**My awesome library**
+
+
+.. code:: python
+
+    def get_report_colors(theme):
+
+        if theme == 'weather':
+            colors = ["blue", "lightblue", "grey"]
+        elif theme == 'traffic':
+            colors = ["red", "orange", "yellow"]
+        else:
+            colors = ["red", "blue", "green"]
+        
+        return colors
+
+
+**Tests**
+
+You can opt to run the tests with the following configuration:
+
+.. code:: python
+
+   def test_get_report_colors():
+
+        assert get_report_colors('weather') == ["blue", "lightblue", "grey"]
+        assert get_report_colors('traffic') == ["red", "orange", "yellow"]
+        assert get_report_colors('other') == ["red", "blue", "green"]
+
+
+Or, if the function is a bit more complex, you split the case in different tests:
+
+.. code:: python
+
+    def test_get_report_colors_weather():
+
+        assert get_report_colors('weather') == ["blue", "lightblue", "grey"]
+
+    def test_get_report_colors_traffic():
+
+        assert get_report_colors('traffic') == ["red", "orange", "yellow"]
+
+    def test_get_report_colors_other():
+
+        assert get_report_colors('other') == ["red", "blue", "green"]
+
+
+In any of the cases, the coverage will be 100% for the function, but the second case will
+be more useful to debug the function.
+
+
+Test Validation GitHub workflow
```

Comment:
I know Vale is not going to like this heading--and I am not sure how it relates to the subsequent text. (There is no mention of GitHub workflows in the content.

Reply:
I changed this to "Continuous Integration and Continuous Deployment (CI/CD) approach". 
Let me know if it suits you.

Reply:
@clatapie That is fine with me.

Reply:
Thank you very much!